File: core/src/main/java/com/rosshambrick/rainorshine/networking/RemoteWeatherStore.java
package com.rosshambrick.rainorshine.networking;

import retrofit.http.GET;
import retrofit.http.Query;
import rx.Observable;

public interface RemoteWeatherStore {
    @GET("/weather")
    Observable<WeatherResponseDto> getWeatherByQuery(@Query("q") String place);

    @GET("/weather")
    Observable<WeatherResponseDto> getWeatherById(@Query("id") long id);
}

File: core/src/main/kotlin/com/rosshambrick/rainorshine/networking/RemoteWeatherStore.kt
package com.rosshambrick.rainorshine.networking

import retrofit.http.GET
import retrofit.http.Query
import rx.Observable

public trait RemoteWeatherStore {
    GET("/weather")
    public fun getWeatherByQuery(Query("q") place: String): Observable<WeatherResponseDto>

    GET("/weather")
    public fun getWeatherById(Query("id") id: Long): Observable<WeatherResponseDto>
}
